numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -393,6 +393,7 @@ _get_cast_safety_from_castingimpl(PyArrayMethodObject *castingimpl,
     Py_DECREF(out_descrs[1]);
     /* NPY_NO_CASTING has to be used for (NPY_EQUIV_CASTING|_NPY_CAST_IS_VIEW) */
     assert(casting != (NPY_EQUIV_CASTING|_NPY_CAST_IS_VIEW));
+    assert(casting != NPY_NO_CASTING);  /* forgot to set the view flag */
```

Comment:
A bit tricky.  But for casts those two asserts are strictly required.

In general, if an `ArrayMethod` does not describe a cast, it make sense to have `NO_CASTING` but `CAST_IS_VIEW` does not make sense.  (E.g. most ufunc loops must do this as default, this is noted in NEP 43.)

Reply:
It makes sense that if no casting is needed, a view should always be possible. But could you update the comment 2 lines above? That said that `NPY_NO_CASTING` has to be used - I guess it really is 
```
/* (NPY_NO_CASTING|_NPY_CAST_IS_VIEW) has to be used for (NPY_EQUIV_CASTING|_NPY_CAST_IS_VIEW) */
```